mautic/mautic
# Some winner criteria not working properly in A/B Tests

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | features  
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | no
| Related developer documentation PR URL | no




#### Description:
There are 2 issues addressed in this PR:
1. The landing page A/B test lowest bounce rate shows highest rate.
2. The email A/B test download rate was not calculated properly.

[//]: # ( As applicable: )
#### Steps to reproduce the landing page A/B test bug:
1. Create landing page with A/B tests variant (Page A & Page B) and one normal page (Page C).
2. Open page (Page A) in private or incognito/private browser window and close the window. This will be considered as bounced.
3. Do step 2 until Page B is visible, do not close window, open normal page (Page C) and then close the window.
4. Page A will be shown as winner (list item with a trophy).

#### Steps to reproduce the email A/B test bug:
1. Create segment email with A/B test. Add some asset link via merge token from builder when saving.
2. Send email to segment.
3. Click asset link in email body (from the inbox).
4. Email A/B test stats and winner won't show.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. After this PR, above bugs should get resolved.

P.S. I have also added a close button on the A/B stats pop up, without that a page reload is required as it blocks from clicking anywhere else in the site.
